### PR TITLE
1905/nojira passion with context without block

### PR DESCRIPTION
**Pull request by jebentier**
### Summary of Changes
* Adding support for using the `with_context` method without a block

### CODE REVIEW

File: lib/contextual_logger.rb [hunk 1]
```diff
@@ -1,26 +1,28 @@
...
+    ContextualLogger::Context::Handler.new(context).set!
   end
 
   def with_context(context)
-    previous_context = Thread.current[THREAD_CONTEXT_NAMESPACE] || {}
-    Thread.current[THREAD_CONTEXT_NAMESPACE] = previous_context.merge(context)
+    context_handler = ContextualLogger::Context::Handler.new(current_context_for_thread.merge(context))
+    context_handler.set!
     yield if block_given?
+    context_handler
```
Comment (alecjacobs5401) [inline lib/contextual_logger.rb]: is there any benefit to returning the context handler here? It breaks the return of anything that happens in the block provided.

For instance, if there is any code that currently does
```rb
return_value = Logger.with_context(context: "logging") { Model.create! }
```
we will no longer get the created model as the response but instead will get the context handler

File: lib/contextual_logger/context/handler.rb [hunk 2]
```diff
@@ -0,0 +1,28 @@
+# frozen_string_literal: true
+
+module ContextualLogger
+  module Context
+    class Handler
```
Comment (alecjacobs5401) [inline lib/contextual_logger/context/handler.rb]: 👏

File: spec/lib/contextual_logger_spec.rb [hunk 3]
```diff
@@ -171,6 +172,79 @@
...
+        severity: 'INFO',
+        timestamp: Time.now,
+        progname: nil
+      }.to_json
+
+      expect_any_instance_of(Logger::LogDevice).to receive(:write).with("#{expected_log_line}\n")
+
+      handler = @logger.with_context(service: 'test_service')
+      expect(@logger.info('this is a test')).to eq(true)
+      handler.reset!
```
Comment (alecjacobs5401) [inline spec/lib/contextual_logger_spec.rb]: This seems like the main use case for returning the handler...maybe only return the handler only if no block is provided?

File: lib/contextual_logger.rb (hunk 1 above)
Comment (jebentier) [reply lib/contextual_logger.rb]: good catch, definitely want to only return if no block was given. will update that

**Pull request merged**